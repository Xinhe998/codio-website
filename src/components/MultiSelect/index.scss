.MultiSelect {
  display: flex;
  flex-direction: column;
  transition: all 0.5s;
}

.MultiSelect__input-container {
  border-radius: 5px;
  border: solid 1px #e4e4e4;
  background-color: #ffffff;
  padding: 1px 5px;
  transition: all 0.5s;
  cursor: text;
  &.focus {
    border: solid 1px #3597ec;
  }
}

.MultiSelect__input {
  outline: 0;
  border: none;
  height: 2.5em;
  width: auto;
  font-size: 13px;
  max-width: 100%;
  line-height: 1.33;
}

.MultiSelect__tags {
  list-style: none;
  padding: 0;
  margin: 0 auto;
  display: inline;
  .tag {
    width: fit-content;
    border-radius: 5px;
    background-color: #e8f0fb;
    font-size: 12px;
    line-height: 1.33;
    text-align: left;
    color: #3597ec;
    padding: 3px 3px 3px 10px;
    margin-top: 5px;
    margin-right: 5px;
    display: inline-table;
    vertical-align: top;
    overflow: hidden;
    svg {
      width: 20px;
      height: 20px;
      color: #3597ec;
      font-weight: bolder;
      vertical-align: middle;
      margin-left: 10px;
      cursor: pointer;
    }
  }
}

.MultiSelect__title {
  font-size: 7px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.14;
  letter-spacing: normal;
  text-align: left;
  color: #263238;
  margin-bottom: 7px;
}

.MultiSelect__dropdown {
  box-shadow: 0 3px 6px 0 rgba(0, 0, 0, 0.16);
  background-color: #ffffff;
  color: #707070;
  padding: 0;
  list-style: none;
  border-radius: 5px;
  padding: 5px 0;
  margin: 5px 0 0;
  max-height: 100px;
  overflow-y: scroll;
  li {
    font-size: 14px;
    padding: 4px 10px;
    cursor: pointer;
    transition: all 0.5s;
    &:hover {
      background: #e8f0fb;
    }
  }
}
