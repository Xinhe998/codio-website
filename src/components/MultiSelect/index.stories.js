import React, { useState } from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { storiesOf } from '@storybook/react';
import { DocsPage, DocsContainer } from '@storybook/addon-docs/blocks';
import StoryRouter from 'storybook-react-router';
import MultiSelect from './index';
import storeFs from '../../reducers';

const store = createStore(storeFs);
const stories = storiesOf('MultiSelect', module).addParameters({
  title: 'MultiSelect',
  component: MultiSelect,
  docs: {
    container: DocsContainer,
    page: DocsPage,
  },
});

const fakeOptions = ['React', 'Vue', 'Angular', 'jQuery', 'CSS', 'HTML'];

stories.addDecorator(StoryRouter()).add('default', () => {
  const Parent = ({ children, ...props }) => {
    const [projectTags, setProjectTags] = useState([]);
    return <div>{children(projectTags, setProjectTags)}</div>;
  };
  return (
    <Parent>
      {(projectTags, setProjectTags) => (
        <Provider store={store}>
          <MultiSelect
            title="類別"
            options={fakeOptions}
            selectedItems={projectTags}
            onChange={setProjectTags}
          />
        </Provider>
      )}
    </Parent>
  );
});
