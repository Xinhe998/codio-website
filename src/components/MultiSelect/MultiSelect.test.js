/* eslint-disable func-names */
import 'babel-polyfill';
import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { shallow, configure, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MultiSelect from './index';

import storeFs from '../../reducers';

const store = createStore(storeFs);

// 以該解析器提供給 Enzyme 做渲染 Component 的設置
configure({ adapter: new Adapter() });

describe('MultiSelect', function () {
  let testComp;

  beforeEach(() => {
    jest.resetModules();
    this.params = {
      title: 'TestMultiSelect',
      options: ['option1', 'option2', 'option3'],
      selectedItems: ['option1'],
      onChange: jest.fn(),
    };

    this.makeSubject = () => {
      const { title, options, selectedItems, onChange } = this.params;

      testComp = mount(
        <Router>
          <Provider store={store}>
            <MultiSelect
              title={title}
              options={options}
              selectedItems={selectedItems}
              onChange={onChange}
            />
          </Provider>
        </Router>,
      );
      return testComp;
    };
  });
  describe('when it has complete data', () => {
    beforeEach(() => {
      this.subject = this.makeSubject();
    });
    it('should be same as snapshot', async () => {
      const wrapper = render(await testComp);
      expect(wrapper).toMatchSnapshot();
    });
    it('should find className', async () => {
      expect(testComp.find('.MultiSelect'));
    });
  });
});
