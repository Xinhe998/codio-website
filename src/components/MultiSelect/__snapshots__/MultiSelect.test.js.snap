// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelect when it has complete data should be same as snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MultiSelect",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "MultiSelect__title",
        },
        "children": Array [
          Object {
            "data": "TestMultiSelect",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "MultiSelect__input-container",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "MultiSelect__tags",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "tag",
                  },
                  "children": Array [
                    Object {
                      "data": "option1",
                      "next": Object {
                        "attribs": Object {
                          "fill": "currentColor",
                          "height": "1em",
                          "stroke": "currentColor",
                          "stroke-width": "0",
                          "viewBox": "0 0 512 512",
                          "width": "1em",
                          "xmlns": "http://www.w3.org/2000/svg",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "d": "M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z",
                            },
                            "children": Array [],
                            "name": "path",
                            "namespace": "http://www.w3.org/2000/svg",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "d": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "d": undefined,
                            },
                          },
                        ],
                        "name": "svg",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "fill": undefined,
                          "height": undefined,
                          "stroke": undefined,
                          "stroke-width": undefined,
                          "viewBox": undefined,
                          "width": undefined,
                          "xmlns": "http://www.w3.org/2000/xmlns/",
                        },
                        "x-attribsPrefix": Object {
                          "fill": undefined,
                          "height": undefined,
                          "stroke": undefined,
                          "stroke-width": undefined,
                          "viewBox": undefined,
                          "width": undefined,
                          "xmlns": "",
                        },
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                    Object {
                      "attribs": Object {
                        "fill": "currentColor",
                        "height": "1em",
                        "stroke": "currentColor",
                        "stroke-width": "0",
                        "viewBox": "0 0 512 512",
                        "width": "1em",
                        "xmlns": "http://www.w3.org/2000/svg",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "d": "M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z",
                          },
                          "children": Array [],
                          "name": "path",
                          "namespace": "http://www.w3.org/2000/svg",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "d": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "d": undefined,
                          },
                        },
                      ],
                      "name": "svg",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "data": "option1",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "fill": undefined,
                        "height": undefined,
                        "stroke": undefined,
                        "stroke-width": undefined,
                        "viewBox": undefined,
                        "width": undefined,
                        "xmlns": "http://www.w3.org/2000/xmlns/",
                      },
                      "x-attribsPrefix": Object {
                        "fill": undefined,
                        "height": undefined,
                        "stroke": undefined,
                        "stroke-width": undefined,
                        "viewBox": undefined,
                        "width": undefined,
                        "xmlns": "",
                      },
                    },
                  ],
                  "name": "li",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "ul",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "aria-busy": "false",
                  "aria-expanded": "false",
                  "aria-haspopup": "true",
                  "aria-owns": "MultiSelect__input-container",
                  "class": "MultiSelect__input",
                  "placeholder": "",
                  "role": "listbox",
                  "size": "2",
                  "value": "",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-busy": undefined,
                  "aria-expanded": undefined,
                  "aria-haspopup": undefined,
                  "aria-owns": undefined,
                  "class": undefined,
                  "placeholder": undefined,
                  "role": undefined,
                  "size": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-busy": undefined,
                  "aria-expanded": undefined,
                  "aria-haspopup": undefined,
                  "aria-owns": undefined,
                  "class": undefined,
                  "placeholder": undefined,
                  "role": undefined,
                  "size": undefined,
                  "value": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "aria-busy": "false",
                "aria-expanded": "false",
                "aria-haspopup": "true",
                "aria-owns": "MultiSelect__input-container",
                "class": "MultiSelect__input",
                "placeholder": "",
                "role": "listbox",
                "size": "2",
                "value": "",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "MultiSelect__tags",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "tag",
                    },
                    "children": Array [
                      Object {
                        "data": "option1",
                        "next": Object {
                          "attribs": Object {
                            "fill": "currentColor",
                            "height": "1em",
                            "stroke": "currentColor",
                            "stroke-width": "0",
                            "viewBox": "0 0 512 512",
                            "width": "1em",
                            "xmlns": "http://www.w3.org/2000/svg",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {
                                "d": "M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z",
                              },
                              "children": Array [],
                              "name": "path",
                              "namespace": "http://www.w3.org/2000/svg",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {
                                "d": undefined,
                              },
                              "x-attribsPrefix": Object {
                                "d": undefined,
                              },
                            },
                          ],
                          "name": "svg",
                          "namespace": "http://www.w3.org/2000/svg",
                          "next": null,
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "fill": undefined,
                            "height": undefined,
                            "stroke": undefined,
                            "stroke-width": undefined,
                            "viewBox": undefined,
                            "width": undefined,
                            "xmlns": "http://www.w3.org/2000/xmlns/",
                          },
                          "x-attribsPrefix": Object {
                            "fill": undefined,
                            "height": undefined,
                            "stroke": undefined,
                            "stroke-width": undefined,
                            "viewBox": undefined,
                            "width": undefined,
                            "xmlns": "",
                          },
                        },
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                      Object {
                        "attribs": Object {
                          "fill": "currentColor",
                          "height": "1em",
                          "stroke": "currentColor",
                          "stroke-width": "0",
                          "viewBox": "0 0 512 512",
                          "width": "1em",
                          "xmlns": "http://www.w3.org/2000/svg",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "d": "M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z",
                            },
                            "children": Array [],
                            "name": "path",
                            "namespace": "http://www.w3.org/2000/svg",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "d": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "d": undefined,
                            },
                          },
                        ],
                        "name": "svg",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": null,
                        "parent": [Circular],
                        "prev": Object {
                          "data": "option1",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "fill": undefined,
                          "height": undefined,
                          "stroke": undefined,
                          "stroke-width": undefined,
                          "viewBox": undefined,
                          "width": undefined,
                          "xmlns": "http://www.w3.org/2000/xmlns/",
                        },
                        "x-attribsPrefix": Object {
                          "fill": undefined,
                          "height": undefined,
                          "stroke": undefined,
                          "stroke-width": undefined,
                          "viewBox": undefined,
                          "width": undefined,
                          "xmlns": "",
                        },
                      },
                    ],
                    "name": "li",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "ul",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-busy": undefined,
                "aria-expanded": undefined,
                "aria-haspopup": undefined,
                "aria-owns": undefined,
                "class": undefined,
                "placeholder": undefined,
                "role": undefined,
                "size": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-busy": undefined,
                "aria-expanded": undefined,
                "aria-haspopup": undefined,
                "aria-owns": undefined,
                "class": undefined,
                "placeholder": undefined,
                "role": undefined,
                "size": undefined,
                "value": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "MultiSelect__input-container",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "MultiSelect__tags",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "tag",
                },
                "children": Array [
                  Object {
                    "data": "option1",
                    "next": Object {
                      "attribs": Object {
                        "fill": "currentColor",
                        "height": "1em",
                        "stroke": "currentColor",
                        "stroke-width": "0",
                        "viewBox": "0 0 512 512",
                        "width": "1em",
                        "xmlns": "http://www.w3.org/2000/svg",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "d": "M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z",
                          },
                          "children": Array [],
                          "name": "path",
                          "namespace": "http://www.w3.org/2000/svg",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "d": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "d": undefined,
                          },
                        },
                      ],
                      "name": "svg",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "fill": undefined,
                        "height": undefined,
                        "stroke": undefined,
                        "stroke-width": undefined,
                        "viewBox": undefined,
                        "width": undefined,
                        "xmlns": "http://www.w3.org/2000/xmlns/",
                      },
                      "x-attribsPrefix": Object {
                        "fill": undefined,
                        "height": undefined,
                        "stroke": undefined,
                        "stroke-width": undefined,
                        "viewBox": undefined,
                        "width": undefined,
                        "xmlns": "",
                      },
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                  Object {
                    "attribs": Object {
                      "fill": "currentColor",
                      "height": "1em",
                      "stroke": "currentColor",
                      "stroke-width": "0",
                      "viewBox": "0 0 512 512",
                      "width": "1em",
                      "xmlns": "http://www.w3.org/2000/svg",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "d": "M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z",
                        },
                        "children": Array [],
                        "name": "path",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "d": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "d": undefined,
                        },
                      },
                    ],
                    "name": "svg",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "data": "option1",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "fill": undefined,
                      "height": undefined,
                      "stroke": undefined,
                      "stroke-width": undefined,
                      "viewBox": undefined,
                      "width": undefined,
                      "xmlns": "http://www.w3.org/2000/xmlns/",
                    },
                    "x-attribsPrefix": Object {
                      "fill": undefined,
                      "height": undefined,
                      "stroke": undefined,
                      "stroke-width": undefined,
                      "viewBox": undefined,
                      "width": undefined,
                      "xmlns": "",
                    },
                  },
                ],
                "name": "li",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "ul",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "aria-busy": "false",
                "aria-expanded": "false",
                "aria-haspopup": "true",
                "aria-owns": "MultiSelect__input-container",
                "class": "MultiSelect__input",
                "placeholder": "",
                "role": "listbox",
                "size": "2",
                "value": "",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-busy": undefined,
                "aria-expanded": undefined,
                "aria-haspopup": undefined,
                "aria-owns": undefined,
                "class": undefined,
                "placeholder": undefined,
                "role": undefined,
                "size": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-busy": undefined,
                "aria-expanded": undefined,
                "aria-haspopup": undefined,
                "aria-owns": undefined,
                "class": undefined,
                "placeholder": undefined,
                "role": undefined,
                "size": undefined,
                "value": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "aria-busy": "false",
              "aria-expanded": "false",
              "aria-haspopup": "true",
              "aria-owns": "MultiSelect__input-container",
              "class": "MultiSelect__input",
              "placeholder": "",
              "role": "listbox",
              "size": "2",
              "value": "",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "MultiSelect__tags",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "tag",
                  },
                  "children": Array [
                    Object {
                      "data": "option1",
                      "next": Object {
                        "attribs": Object {
                          "fill": "currentColor",
                          "height": "1em",
                          "stroke": "currentColor",
                          "stroke-width": "0",
                          "viewBox": "0 0 512 512",
                          "width": "1em",
                          "xmlns": "http://www.w3.org/2000/svg",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "d": "M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z",
                            },
                            "children": Array [],
                            "name": "path",
                            "namespace": "http://www.w3.org/2000/svg",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "d": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "d": undefined,
                            },
                          },
                        ],
                        "name": "svg",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "fill": undefined,
                          "height": undefined,
                          "stroke": undefined,
                          "stroke-width": undefined,
                          "viewBox": undefined,
                          "width": undefined,
                          "xmlns": "http://www.w3.org/2000/xmlns/",
                        },
                        "x-attribsPrefix": Object {
                          "fill": undefined,
                          "height": undefined,
                          "stroke": undefined,
                          "stroke-width": undefined,
                          "viewBox": undefined,
                          "width": undefined,
                          "xmlns": "",
                        },
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                    Object {
                      "attribs": Object {
                        "fill": "currentColor",
                        "height": "1em",
                        "stroke": "currentColor",
                        "stroke-width": "0",
                        "viewBox": "0 0 512 512",
                        "width": "1em",
                        "xmlns": "http://www.w3.org/2000/svg",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "d": "M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z",
                          },
                          "children": Array [],
                          "name": "path",
                          "namespace": "http://www.w3.org/2000/svg",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "d": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "d": undefined,
                          },
                        },
                      ],
                      "name": "svg",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "data": "option1",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "fill": undefined,
                        "height": undefined,
                        "stroke": undefined,
                        "stroke-width": undefined,
                        "viewBox": undefined,
                        "width": undefined,
                        "xmlns": "http://www.w3.org/2000/xmlns/",
                      },
                      "x-attribsPrefix": Object {
                        "fill": undefined,
                        "height": undefined,
                        "stroke": undefined,
                        "stroke-width": undefined,
                        "viewBox": undefined,
                        "width": undefined,
                        "xmlns": "",
                      },
                    },
                  ],
                  "name": "li",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "ul",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "aria-busy": undefined,
              "aria-expanded": undefined,
              "aria-haspopup": undefined,
              "aria-owns": undefined,
              "class": undefined,
              "placeholder": undefined,
              "role": undefined,
              "size": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "aria-busy": undefined,
              "aria-expanded": undefined,
              "aria-haspopup": undefined,
              "aria-owns": undefined,
              "class": undefined,
              "placeholder": undefined,
              "role": undefined,
              "size": undefined,
              "value": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "MultiSelect__title",
          },
          "children": Array [
            Object {
              "data": "TestMultiSelect",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
